### DISCUSSION TITLE
How can I install this repo as the default TTS?

### DISCUSSION THREAD

**Discussion by Sotiris-Bekiaris** [idiap/coqui-ai-TTS#67] [Q&A] [answered]
For example I want to use this
```python
from TTS.tts.configs.xtts_config import XttsConfig
from TTS.tts.models.xtts import Xtts
from TTS.utils.generic_utils import get_user_data_dir
from TTS.utils.manage import ModelManager
```

what should my requirements.txt look like in order to install your forked version? Because I am trying to use XTTS 2 from the original repo and I get an error that it seems you already fixed it. 

I am talking about [this](https://github.com/idiap/coqui-ai-TTS/issues/31) one.

Now I am setting up a conda env but I cant seem to install correctly your repo.

I did this but I am not sure it is correct

```
pip install git+https://github.com/idiap/coqui-ai-TTS.git -vv
```

and I get this after I run my code

```
    from TTS.utils.generic_utils import get_user_data_dir
ImportError: cannot import name 'get_user_data_dir' from 'TTS.utils.generic_utils' (/opt/anaconda3/envs/xtts/lib/python3.9/site-packages/TTS/utils/generic_utils.py)

```

**Comment by Sotiris-Bekiaris** [comment]
For context I run 

```
pip list
```

and I can see that these are installed

```
coqui-tts               0.24.1
coqui-tts-trainer       0.1.4
```


also [this](https://github.com/huggingface/transformers/issues/31040#issuecomment-2134496460) is what I am getting from the original repo. (this is where I found your repo from as well)

**Reply by Sotiris-Bekiaris** [reply]
I feel like the fix that was merged in the dev branch [here](https://github.com/idiap/coqui-ai-TTS/pull/46) has not yet been released since I can see that the latest release is at May 29 and includes only [this](https://github.com/idiap/coqui-ai-TTS/pull/33) workaround.

Am I missing something? 

Shouldn't it work though with the workaround?

Anyways I think I will try to find a way to install the HEAD of dev branch, see if that works...

**Comment by Sotiris-Bekiaris** [comment]
> ```
>     from TTS.utils.generic_utils import get_user_data_dir
> ImportError: cannot import name 'get_user_data_dir' from 'TTS.utils.generic_utils' (/opt/anaconda3/envs/xtts/lib/python3.9/site-packages/TTS/utils/generic_utils.py)
> ```

Alright I took a look in your code and I saw that you use the get_user_data_dir like that:

```python
from trainer.io import get_user_data_dir
```

replacing that import worked and resolved my issue, but now I get once again this error:

```
TypeError: isin() received an invalid combination of arguments - got (test_elements=int, elements=Tensor, ), but expected one of:
 * (Tensor elements, Tensor test_elements, *, bool assume_unique = False, bool invert = False, Tensor out = None)
 * (Number element, Tensor test_elements, *, bool assume_unique = False, bool invert = False, Tensor out = None)
 * (Tensor elements, Number test_element, *, bool assume_unique = False, bool invert = False, Tensor out = None)
```

I am sorry if this is getting a bit chaotic. 

Here is a screenshot of my `pip list` just to be sure

![image](https://github.com/user-attachments/assets/59881980-15f7-4150-9eb9-35dacfd53a04)

**Comment by eginhard** [comment, marked as answer]
Hey @Sotiris-Bekiaris ! Looks like you already figured out most things.
- Yes, the import should now be `from trainer.io import get_user_data_dir` (there is some duplicate code in the TTS and Trainer packages that I'm cleaning up to make maintenance easier)
- The code in the current `dev` head (which you're already getting with your `pip install git+...`) with the fix from #46 only works for `transformers==4.42.*` at the moment, so you could manually install a matching `transformers` version.
- You can follow #65 to get updated when a fix for `transformers>=4.43.*` is available. There is a patch posted there, but I haven't had time to test and merge it yet. I'll make a new proper release after that as well.

**Reply by Sotiris-Bekiaris** [reply]
Thank you so much for the answer! Installing `transformers==4.42.4` solved it. Also thank you for the contribution to OSS!
